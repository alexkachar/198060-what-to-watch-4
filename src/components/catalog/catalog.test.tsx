import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import Catalog from './catalog';
import MOCK_MOVIES from '../../test-data/mock-movies';
import MOCK_GENRES from '../../test-data/mock-genres';

const SELECTED_GENRE = `Action`;

it(`Catalog renders correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <Catalog
            movies={MOCK_MOVIES}
            genres={MOCK_GENRES}
            selectedGenre={SELECTED_GENRE}
            onGenreSelect={jest.fn()}
          />
        </BrowserRouter>, {
          createNodeMock: () => document.createElement(`div`)
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
