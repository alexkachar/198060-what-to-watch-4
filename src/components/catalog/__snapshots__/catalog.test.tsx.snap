// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders correctly 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item null"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Action
      </a>
    </li>
    <li
      className="catalog__genres-item null"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Adventure
      </a>
    </li>
    <li
      className="catalog__genres-item null"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Comedy
      </a>
    </li>
    <li
      className="catalog__genres-item null"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Crime
      </a>
    </li>
    <li
      className="catalog__genres-item null"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Drama
      </a>
    </li>
    <li
      className="catalog__genres-item null"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Fantasy
      </a>
    </li>
    <li
      className="catalog__genres-item null"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Thriller
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        href="/film/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            muted={true}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            poster="img/the-grand-budapest-hotel.jpg"
            src="https://some-link"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/film/1"
          onClick={[Function]}
          style={
            Object {
              "color": "#c9b37e",
            }
          }
        >
          The Grand Budapest Hotel
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        href="/film/2"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            muted={true}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/film/2"
          onClick={[Function]}
          style={
            Object {
              "color": "#c9b37e",
            }
          }
        >
          Moonrise Kingdom
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <a
        href="/film/15"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            muted={true}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
            width="280"
          />
        </div>
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/film/15"
          onClick={[Function]}
          style={
            Object {
              "color": "#c9b37e",
            }
          }
        >
          Fantastic Beasts: The Crimes of Grindelwald
        </a>
      </h3>
    </article>
  </div>
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;
